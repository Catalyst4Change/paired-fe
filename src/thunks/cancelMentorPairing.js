import { fetchData } from '../utils';
import { setError, setLoading, cancelPairing } from '../actions';
import * as gql from '../queries';

export const cancelMentorPairing = (pairingId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.cancelMentorPairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(cancelPairing(pairingId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
