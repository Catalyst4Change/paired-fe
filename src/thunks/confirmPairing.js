import { fetchData } from '../utils';
import { setError, setLoading, addToSchedule, clearAvailPairings } from '../actions';
import * as gql from '../queries';

export const confirmPairing = (pairingId, paireeId, notes) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const query = gql.updatePairing(pairingId, paireeId, notes);
      const response = await fetchData(query);
      const pairing = await response.updatePairing;
      dispatch(addToSchedule(pairing));
      dispatch(clearAvailPairings());
    } catch(error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
};
