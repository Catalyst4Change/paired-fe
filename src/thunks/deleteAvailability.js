import { fetchData } from "../utils";
import { setError, setLoading } from '../actions';
import * as gql from '../queries';

export const deleteAvailability = (userId) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const mutation = gql.deletePairings(userId);
      await fetchData(mutation);
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
};