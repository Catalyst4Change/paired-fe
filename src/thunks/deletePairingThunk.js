import { fetchData } from '../utils';
import { setError, setLoading, deletePairing } from '../actions';
import * as gql from '../queries';

export const deletePairingThunk = (pairingId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.deletePairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(deletePairing(pairingId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
