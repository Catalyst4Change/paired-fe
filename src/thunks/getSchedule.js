import { fetchData } from '../utils';
import { setError, setLoading, setSchedule } from '../actions';
import * as gql from '../queries';

export const getSchedule = (id) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const pairingQuery = gql.getUserPairings(id);
      const pairingRes = await fetchData(pairingQuery);
      dispatch(setSchedule(pairingRes.getUserPairings));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
