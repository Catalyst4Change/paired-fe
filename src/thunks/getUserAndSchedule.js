import { fetchData } from '../utils';
import { setError, setLoading, setSchedule, setUser } from '../actions';
import * as gql from '../queries';

export const getUserAndSchedule = (userID) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const userQuery = gql.getUser(userID);
      const userRes = await fetchData(userQuery);
      const pairingQuery = gql.getUserPairings(userRes.getUser.id);
      const pairingRes = await fetchData(pairingQuery);
      dispatch(setUser(userRes.getUser));
      dispatch(setSchedule(pairingRes.getUserPairings));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
