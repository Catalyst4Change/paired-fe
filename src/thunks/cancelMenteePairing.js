import { fetchData } from '../utils';
import { setError, setLoading } from '../actions';
import { getSchedule } from './getSchedule';
import * as gql from '../queries';

export const cancelMenteePairing = (pairingId, userId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.cancelMenteePairing(pairingId);
      await fetchData(query);
      dispatch(setLoading(false));
      dispatch(getSchedule(userId));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};
