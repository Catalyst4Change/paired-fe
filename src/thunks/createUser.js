import { fetchData } from '../utils';
import { setError, setLoading, setUser } from '../actions';
import { getSchedule } from './getSchedule';
import * as gql from '../queries';
import * as helpers from '../helpers';

export const createUser = (userInfo, availabilities) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const userQuery = gql.createUser(userInfo);
      const { user } = await fetchData(userQuery);
      const startDate = 'Mon Apr 29 2019';
      const inning = helpers.getDatesToDisplayFancy(startDate, 29);
      const unformattedPairings = helpers.createPairingsForQuery(
        availabilities,
        inning,
        user.id
      );
      const formattedPairings = helpers.formatPairingsForQuery(unformattedPairings)
      const pairingQuery = gql.createPairings(formattedPairings);
      await fetchData(pairingQuery);
      dispatch(setUser(user));
      dispatch(getSchedule(user.id));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};




