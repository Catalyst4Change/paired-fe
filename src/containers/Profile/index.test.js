import React from 'react';
import { shallow } from 'enzyme';
import { Profile, mapDispatchToProps } from './index';
import * as thunk from '../../thunks/createUser';

describe('Profile', () => {
  describe('Profile component', () => {
    let wrapper;
    const mockCreateUser = jest.fn();
    const mockProps = {
      createUser: mockCreateUser,
      email: 'Hill@gmail.com',
      firebaseID: 'bAzN8PSfQSPL9rg4pMhTha8zewn2',
      image: 'https://avatars1.githubusercontent.com/u/49459620?v=4',
      name: 'Hillary'
    };
    
    beforeEach(() => {
      wrapper = shallow(<Profile {...mockProps} />);
      window.fetch = jest.fn();
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have a default state', () => {
      const expected = {
        name: mockProps.name,
        email: mockProps.email,
        program: '',
        module: '',
        pronouns: '',
        slack: '',
        skill1: '',
        skill2: '',
        availabilities: [
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
        ]
      };
      expect(wrapper.state()).toEqual(expected);
    });

    it('should return true if the user has not entered info for program, module, email, name, pronouns, or slack', () => {
      const expected = true;
      const result = wrapper.instance().checkDropdowns();
      expect(result).toEqual(expected);
    });

    it('should handleSubmit when the form is submitted', async () => {
      const instance = wrapper.instance();
      jest.spyOn(instance, 'handleSubmit');
      wrapper.instance().forceUpdate();
      wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
      expect(instance.handleSubmit).toHaveBeenCalled();
    });

    it('should toggle true/false the correct index in availabilities when handleClick is called', () => {
      const mockEvent = { preventDefault: jest.fn() };
      const mockIndex = 2;
      const mockInitialAvailability = [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
      ];
      const expected = [
        false,
        false,
        true,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
      ];
      expect(wrapper.state('availabilities')).toEqual(mockInitialAvailability);
      wrapper.instance().handleClick(mockEvent, mockIndex);
      expect(wrapper.state('availabilities')).toEqual(expected);
    });

    it('should call handleChange when a user types in the input', () => {
      const mockSlack = '@tiffany';
      const mockEvent = {
        target: { name: 'slack', value: mockSlack }
      };
      const initialState = '';
      expect(wrapper.state('slack')).toEqual(initialState);
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('slack')).toEqual(mockSlack);
    });
  });

  describe('mapDispatchToProps', () => {
    window.fetch = jest.fn();
    it('should call dispatch with createUser thunk when createUser is called', () => {
      const mockDispatch = jest.fn();
      thunk.createUser = jest.fn();
      const actionToDispatch = thunk.createUser({}, []);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.createUser({}, []);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  })
});
