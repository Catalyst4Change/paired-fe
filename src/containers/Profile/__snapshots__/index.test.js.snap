// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Profile component should match the snapshot 1`] = `
<form
  className="Profile--form"
  onSubmit={[Function]}
>
  <h2>
    Please complete your profile
  </h2>
  <div
    className="Profile--div"
  >
    <div
      className="Profile--div--flex"
    >
      <label
        htmlFor="name"
      >
        Provide your name
        <span>
          *
        </span>
      </label>
      <input
        name="name"
        onChange={[Function]}
        value="Hillary"
      />
      <label
        htmlFor="email"
      >
        Provide your email
        <span>
          *
        </span>
      </label>
      <input
        name="email"
        onChange={[Function]}
        value="Hill@gmail.com"
      />
      <label
        htmlFor="slack"
      >
        Provide your slack handle
        <span>
          *
        </span>
      </label>
      <input
        name="slack"
        onChange={[Function]}
        placeholder="@"
        value=""
      />
    </div>
    <Dropdown
      handleChange={[Function]}
      label="Pronouns"
      options={
        Array [
          "she/her",
          "he/him",
          "they/them",
          "ze/zir",
          "prefer not to answer",
        ]
      }
      required={true}
    />
    <Dropdown
      handleChange={[Function]}
      label="Program"
      options={
        Array [
          "FE",
          "BE",
        ]
      }
      required={true}
    />
    <Dropdown
      handleChange={[Function]}
      label="Module"
      options={
        Array [
          1,
          2,
          3,
          4,
        ]
      }
      required={true}
    />
    <h4>
      Select skills that you feel comfortable helping others with (optional)
    </h4>
    <Dropdown
      handleChange={[Function]}
      label="Skill 1"
      options={
        Array [
          "grid",
          "flexbox",
          "mythical creatures",
          "react",
          "jsFun",
          "rspec",
          "rails",
          "ruby",
          "jest/enzyme",
          "docker",
          "lightning talks",
          "javascript",
          "css",
          "active record",
          "SQL",
          "sinatra",
        ]
      }
    />
    <Dropdown
      handleChange={[Function]}
      label="Skill 2"
      options={
        Array [
          "grid",
          "flexbox",
          "mythical creatures",
          "react",
          "jsFun",
          "rspec",
          "rails",
          "ruby",
          "jest/enzyme",
          "docker",
          "lightning talks",
          "javascript",
          "css",
          "active record",
          "SQL",
          "sinatra",
        ]
      }
    />
  </div>
  <button
    className="Profile--button"
    disabled={true}
  >
    Submit
  </button>
</form>
`;
